doc/guides/investigating_native_memory_leak.md

```diff
@@ -0,0 +1,441 @@
+ 11   assert(status == napi_ok);
+ 12   for (int i=0; i< 1000; i++) {
+ 13     malloc_holder = malloc(1000);  <<<<<< Yup this is where we are allocating the memory that is not freed
+ 14   }
+ 15   return world;
+ 16 }
+```
+
+### Node.js binary
+
+If the leak is not in an addon and in the Node.js binary itself you may need
+to compile node yourself and turn on debug symbols. Looking at this entry
+reported by valgrind, with the release binary we see:
+
+```bash
+74== 304 bytes in 1 blocks are possibly lost in loss record 27 of 35
+==4174==    at 0x4C31B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
+==4174==    by 0x40134A6: allocate_dtv (dl-tls.c:286)
+==4174==    by 0x40134A6: _dl_allocate_tls (dl-tls.c:530)
+==4174==    by 0x5987227: allocate_stack (allocatestack.c:627)
+==4174==    by 0x5987227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
+==4174==    by 0xAAF9DC: node::inspector::Agent::Start(std::string const&, node::DebugOptions const&, std::shared_ptr<node::HostPort>, bool) (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==4174==    by 0x9A8BE7: node::Environment::InitializeInspector(std::unique_ptr<node::inspector::ParentInspectorHandle, std::default_delete<node::inspector::ParentInspectorHandle> >) (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==4174==    by 0xA1C9A5: node::NodeMainInstance::CreateMainEnvironment(int*) (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==4174==    by 0xA1CB42: node::NodeMainInstance::Run() (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==4174==    by 0x9ACB67: node::Start(int, char**) (in /home/user1/valgrind/node-v12.14.1-linux-x64/bin/node)
+==4174==    by 0x5BBFB96: (below main) (libc-start.c:310)
+```
+
+This gives us some information of where to look (node::inspector::Agent::Start)
+but not where in that function. We get more information that you might expect
+(or see by default with addons) because the Node.js binary exports many of
+its symbols using -rdynamic so that they can be used by addons. If the stack
+gives you enough information to track down where the leak is great,
+otherwise the next step is to compile a Node.js binary with debugging enabled.
+
+To get additional information with valgrind:
+
+* check out the Node.js source corresponding to the release that you
+  want to debug. For example
+```bash
+git clone https://github.com/nodejs/node.git
+git checkout -b v12.14.1
+```
+* compile with debug enabled (for addional info see [building a debug build](https://github.com/nodejs/node/blob/v12.14.1/BUILDING.md#building-a-debug-build)
+```bash
+./configure --debug
+make  -j4
+```
+* make sure to run with your compiled debug version of Node.js. Having used
+ `./configure --debug`, two binaries will have been built when make was run.
+  You must use the one which is in .../out/Debug. Add that version of the
+  binary to your path. For example:
+
+```bash
+export PATH=/home/user1/node/out/Debug/:$PATH
```

Comment:
I don't think this is necessary. You can just specify the full path on the command line instead, which will probably lead to less confusion about which binary is being used.

Reply:
I agree with @mscdex – that being said, if you do want to include commands along these lines, please use quotes around the right-hand side

Reply:
k removed